core/lib/contracts_loader/src/lib.rs

```diff
@@ -0,0 +1,135 @@
+use zksync_config::configs::contracts::{
+    chain::ChainContracts,
+    ecosystem::{EcosystemCommonContracts, L1SpecificContracts},
+    ChainSpecificContracts,
+};
+use zksync_contracts::{bridgehub_contract, state_transition_manager_contract};
+use zksync_eth_client::{CallFunctionArgs, EthInterface};
+use zksync_types::{
+    ethabi::{Contract, Token},
+    settlement::SettlementMode,
+    Address, L2ChainId,
+};
+
+/// Load contacts specific for each settlement layer, using bridgehub contract
+pub async fn load_settlement_layer_contracts(
+    sl_client: &dyn EthInterface,
+    bridgehub_address: Address,
+    l2_chain_id: L2ChainId,
+    multicall3: Option<Address>,
+) -> anyhow::Result<Option<ChainSpecificContracts>> {
+    let gateway_diamond_proxy: Address =
+        CallFunctionArgs::new("getZKChain", Token::Uint(l2_chain_id.as_u64().into()))
+            .for_contract(bridgehub_address, &bridgehub_contract())
+            .call(sl_client)
+            .await?;
+
+    if gateway_diamond_proxy.is_zero() {
+        return Ok(None);
+    }
+
+    let ctm_address =
+        CallFunctionArgs::new("chainTypeManager", Token::Uint(l2_chain_id.as_u64().into()))
+            .for_contract(bridgehub_address, &bridgehub_contract())
+            .call(sl_client)
+            .await?;
+
+    let server_notifier_addr = CallFunctionArgs::new("serverNotifierAddress", ())
+        .for_contract(ctm_address, &state_transition_manager_contract())
+        .call(sl_client)
+        .await
+        .map(|a: Address| if a.is_zero() { None } else { Some(a) })?;
+
+    let validator_timelock_addr = CallFunctionArgs::new("validatorTimelock", ())
+        .for_contract(ctm_address, &state_transition_manager_contract())
```

Comment:
this will also not work fine for chains that have not yet upgraded to v26. For v25->v26 upgrade we used a trick where:
- if the chain has the same version as on CTM, we use the CTM's timelock
- we take it from config otherwise

not sure if the code in this PR is compatible with it

Reply:
do we have any chains, where i can verify it?